### PR TITLE
DOC: prep 1.9.0 release notes

### PR DESCRIPTION
**Pull request by tylerjereddy**
* transcribe entries from the [wiki page](https://github.com/scipy/scipy/wiki/Release-note-entries-for-SciPy-1.9.0) (the history of this page
will be used to update again before branching)
* draft in results from author/list scripts
* some initial manual curation of the author list via `.mailmap` adjustments
* manually update the notes based on merged PRs with `enhancement` label (for
those cases where it struck me as pertinent to mention)
* the "highlights" are still open to adjustment of course; simply
copied from the wiki for now--`meson` comes to mind depending on where
we land near the end of the week?

### CODE REVIEW

File: doc/release/1.9.0-notes.rst [hunk 1]
```diff
@@ -70,60 +145,806 @@ New features
...
+- ``lloyd_centroidal_voronoi_tessellation`` has been added to allow improved
+sample distributions via iterative application of Voronoi diagrams and
+centering operations
 
 Other
 -----
 
+A draft developer CLI is available for SciPy, leveraging the ``doit``,
+``click`` and ``rich-click`` tools. For more details, see
+[gh-15959](https://github.com/scipy/scipy/pull/15959).
```
Comment (tylerjereddy) [inline doc/release/1.9.0-notes.rst:229]: Maybe we can fuse the two "Other" themed sections?

**Comment by tylerjereddy** [comment]
CI correctly highlights a bunch of doc build warnings in the new release notes.

File: doc/release/1.9.0-notes.rst [hunk 2]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
 Highlights of this release
 **************************
 
+- Added `scipy.optimize.milp`, new function for mixed-integer linear
+  programming.
+- Added `scipy.stats.fit` for fitting discrete and continuous distributions
+  to data.
+- Added new spline based interpolation methods for
+  `scipy.interpolate.RegularGridInterpolator`.
```
Comment (ev-br) [inline doc/release/1.9.0-notes.rst:34]: Maybe rephrase to read "Tensor-product spline interpolation modes were added to `scipy.interpolate.RegularGridInterpolator`" or some such.

File: doc/release/1.9.0-notes.rst [hunk 3]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
...
 `scipy.interpolate` improvements
 ================================
-
+- Speed up the ``RBFInterpolator`` evaluation with high dimensional
+interpolants.
+- Added new spline based interpolation methods for
+`scipy.interpolate.RegularGridInterpolator` and its tutorial.
+- `scipy.interpolate.RegularGridInterpolator` and `scipy.interpolate.interpn`
+now accept descending ordered points.
+- ``RegularGridInterpolator`` now handles length-1 grid axes
```
Comment (ev-br) [inline doc/release/1.9.0-notes.rst:69]: Now that gh-16220 is in, let's add

- `BivariateSpline`s have grown a new method `partial_derivative` which constructs a new spline object representing a derivative of an original spline. This mirrors the corresponding functionality for univariate splines, `splder` and `BSpline.derivative`, and can substantially speed up repeated evaluation of derivatives.

**Review by ev-br — COMMENTED** [review]
Two small tweaks --- let me know if you prefer to have them in the wiki  instead Tyler.

File: doc/release/1.9.0-notes.rst [hunk 4]
```diff
@@ -70,60 +145,806 @@ New features
...
+of `scipy.optimize.linprog` are deprecated. Methods ``highs``, ``highs-ds``,
+or ``highs-ipm`` should be used in new code.
+- Support for non-numeric arrays has been deprecated from ``stats.mode``.
+``pandas.DataFrame.mode`` can be used instead.
+- The ``extradoc`` keyword of ``rv_continuous``, ``rv_discrete`` and
+``rv_sample`` now raises a DeprecationWarning, after having been deprecated in
+documentation only for a long time already.
+
+********************
+Expired Deprecations
```
Comment (tupui) [inline doc/release/1.9.0-notes.rst:253]: @h-vetinari @j-bowhay FYI if there are other entries to add, please add them here as suggestion so that it's easier for Tyler.

File: doc/release/1.9.0-notes.rst [hunk 5]
```diff
@@ -70,60 +145,806 @@ New features
...
+- added ``qrvs`` method to ``NumericalInversePolynomial`` to match the
+situation for ``NumericalInverseHermite``
+
+- faster random variate generation for ``gennorm``
+
+- ``lloyd_centroidal_voronoi_tessellation`` has been added to allow improved
+sample distributions via iterative application of Voronoi diagrams and
+centering operations
 
 Other
```
Comment (tupui) [inline doc/release/1.9.0-notes.rst:224]:
```suggestion
centering operations.

- Add `scipy.stats.qmc.PoissonDisk` to sample using the Poisson disk sampling method. It guarantees that samples are separated from each other by a given ``radius``.

Other
```

File: doc/release/1.9.0-notes.rst (hunk 4 above)
Comment (h-vetinari) [reply doc/release/1.9.0-notes.rst:253]: I've tried to keep the wiki up-to-date with the issues in https://github.com/scipy/scipy/issues/15765 as they were closed. To the best of my knowledge, the list as below is complete.

Comment (tupui) [reply doc/release/1.9.0-notes.rst:253]: Thanks for confirming.

File: doc/release/1.9.0-notes.rst [hunk 6]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- A new global optimizer (DIviding RECTangles algorithm)
+  `scipy.optimize.direct`
+
 
 ************
 New features
 ************
 
+Lazy access to subpackages
+==========================
```
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:44]: I'd say that this doesn't belong here, it should go in the `Other changes` section, like the build system changes. `New features` is for subpackages only.

File: doc/release/1.9.0-notes.rst [hunk 7]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+
+Before this release, all subpackages of SciPy (`cluster`, `fft`, `ndimage`,
+etc.) had to be explicitly imported. Now, these subpackages are lazily loaded
+as soon as they are accessed, so that the following is possible:
```
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:48]: add "(if desired for interactive use, it's not actually recommended for code, see `:ref:`scipy-api`).

File: doc/release/1.9.0-notes.rst [hunk 8]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
...
 **************************
 
+- Added `scipy.optimize.milp`, new function for mixed-integer linear
+  programming.
+- Added `scipy.stats.fit` for fitting discrete and continuous distributions
+  to data.
+- Added new spline based interpolation methods for
+  `scipy.interpolate.RegularGridInterpolator`.
+- A new global optimizer (DIviding RECTangles algorithm)
+  `scipy.optimize.direct`
```
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:36]: I have the Meson release note changes in the PR to change to it by default - but not a highlight. Would be good to add that here indeed.

File: doc/release/1.9.0-notes.rst [hunk 9]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
...
 ===========================
-
+- `scipy.linalg.expm` now accepts nD arrays. It's speed is also improved.
+- Minimum required LAPACK version is bumped to ``3.7.1``
 
 `scipy.ndimage` improvements
 ============================
 
+`scipy.fft` improvements
+============================
```
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:80]: minor: underline length needs to match title length

File: doc/release/1.9.0-notes.rst [hunk 10]
```diff
@@ -26,42 +26,117 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- The new window function `scipy.signal.windows.kaiser_bessel_derived` was
+  added to compute the Kaiser-Bessel derived window.
+- single-precision ``hilbert`` operations are now faster as a result of more
+  consistent ``dtype`` handling
 
 `scipy.sparse` improvements
 ===========================
-
-
+- Add a `copy` parameter to `scipy.sparce.csgraph.laplacian`. Using inplace
```
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:119]: formatting: double backticks for `copy`

Comment (rgommers) [reply doc/release/1.9.0-notes.rst:119]: and same for `copy=False` below

File: doc/release/1.9.0-notes.rst [hunk 11]
```diff
@@ -70,60 +145,806 @@ New features
...
+implementation is based. "S0" has the advantage that ``delta`` and ``gamma``
+are proper location and scale parameters. With ``delta`` and ``gamma`` fixed,
+the location and scale of the resulting distribution remain unchanged as
+``alpha`` and ``beta`` change. This is not the case for the default "S1"
+parametrization. Finally, more options have been exposed to allow users to
+trade off between runtime and accuracy for both the default and FFT methods of
+pdf and cdf calculation. More information can be found in the documentation
+here (to be linked).
+
+- Added `scipy.stats.fit` for fitting discrete and continuous distributions to
```
Comment (rgommers) [inline doc/release/1.9.0-notes.rst:189]: Just noticing this now, `fit` is a _very_ generic name. I can't remember it being proposed/discussed. Will comment on gh-15436.

File: doc/release/1.9.0-notes.rst (hunk 1 above)
Comment (rgommers) [reply doc/release/1.9.0-notes.rst:229]: Yes, good idea.

File: doc/release/1.9.0-notes.rst [hunk 12]
```diff
@@ -70,60 +145,806 @@ New features
...
 *************
 
+- The SciPy contributor guide has been reorganized and updated
+(see [#15947](https://github.com/scipy/scipy/pull/15947) for details).
+
+- QUADPACK Fortran routines in `scipy.integrate`, which power
+`scipy.integrate.quad`, have been marked as `recursive`. This should fix rare
+issues in multivariate integration (`nquad` and friends) and obviate the need
+for compiler-specific compile flags (`/recursive` for ifort etc). Please file
+an issue if this change turns out problematic for you.
```
Comment (ev-br) [inline doc/release/1.9.0-notes.rst:333]: Also FITPACK routines in `scipy.interpolate`, which power `splrep`, `splev` etc, and `*UnivariateSpline` and `*BivariateSpline` classes.

File: doc/release/1.9.0-notes.rst [hunk 13]
```diff
@@ -70,60 +145,806 @@ New features
...
+implementation and thus was accepting sparse arrays. Now it only works with
+nDarrays. For sparse usage, `scipy.sparse.linalg.expm` needs to be used
+explicitly.
+- The definition of `scipy.stats.circvar` has reverted to the one that is
+standard in the literature; note that this is not the same as the square of
+`scipy.stats.circstd`.
+- Remove inheritance to `QMCEngine` in `MultinomialQMC` and
+`MultivariateNormalQMC`. It removes the methods `fast_forward` and `reset`.
+- Init of `MultinomialQMC` now require the number of trials with `n_trials`.
+Hence, `MultinomialQMC.random` output has now the correct shape ``(n, pvals)``.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:319]: Email [here](https://mail.python.org/archives/list/scipy-dev@python.org/thread/QEMS6VZSBHTRB6OVSDBULJNNOE7PFAOV/#QEMS6VZSBHTRB6OVSDBULJNNOE7PFAOV), discussion [here](https://github.com/scipy/scipy/pull/15945). This was definitely a bug.

File: doc/release/1.9.0-notes.rst [hunk 14]
```diff
@@ -70,60 +145,806 @@ New features
...
+* `#16081 <https://github.com/scipy/scipy/issues/16081>`__: CI, MAINT: minor refguide failure with stats.describe
+* `#16121 <https://github.com/scipy/scipy/issues/16121>`__: DOC: scipy.interpolate.RegularGridInterpolator and interpn works...
+* `#16162 <https://github.com/scipy/scipy/issues/16162>`__: BUG: curve_fit gives wrong results with Pandas float32
+* `#16171 <https://github.com/scipy/scipy/issues/16171>`__: BUG: scipy.stats.multivariate_hypergeom.rvs raises ValueError...
 
 ***********************
 Pull requests for 1.9.0
 ***********************
 
+* `#354 <https://github.com/scipy/scipy/pull/354>`__: Faster expm
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:615]:
```suggestion
```
This wasn't actually merged; it was superseded. Perhaps the code that generates this list could use a little tweak? Not too important, though.

File: doc/release/1.9.0-notes.rst [hunk 15]
```diff
@@ -70,60 +145,806 @@ New features
...
+
+- Add a `integers` method to `scipy.stats.qmc.QMCEngine`. It allows sampling
+integers using any QMC sampler.
+
+- improved the fit speed and accuracy of ``stats.pareto``
+
+- added ``qrvs`` method to ``NumericalInversePolynomial`` to match the
+situation for ``NumericalInverseHermite``
+
+- faster random variate generation for ``gennorm``
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:218]:
```suggestion
- faster random variate generation for ``gennorm`` and ``nakagami``
```
gh-15360

File: doc/release/1.9.0-notes.rst [hunk 16]
```diff
@@ -70,60 +145,806 @@ New features
...
+[#15373](https://github.com/scipy/scipy/issues/15373). In addition, the
+documentation for `scipy.stats.combine_pvalues` has been expanded and improved.
+
+- Unlike other reduction functions, ``stats.mode`` didn't consume the axis
+being operated on and failed for negative axis inputs. Both the bugs have been
+fixed. Note that ``stats.mode`` will now consume the input axis and return an
+ndarray with the ``axis`` dimension removed.
+
+- Replaced implementation of `scipy.stats.ncf` with the implementation from
+Boost.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:203]:
```suggestion
Boost for improved reliability.
```

**Comment by tylerjereddy** [comment]
I think I addressed all the comments in the PR so far, and `make html-scipyorg` passes locally (using the old `setuptools` approach). I've also included a fix to `gh_lists.py` that ignores PRs that were never actually merged (i.e., the closed PRs @mdhaber pointed out are no longer appended now). You can see the revisions in the second commit in the PR, `307443`. I also rebased to pull in the latest `meson` stuff just merged.

There will still be one or two more merges to include here and I'll check the wiki again before finalizing here/branching. I'll check the html version of the relnotes produced locally as well if I can.

**Comment by tylerjereddy** [comment]
From the html preview locally, I'd say one or two sections below the fold look incorrectly formatted (`Expired Deprecations`, `Lazy access to subpackages`; the former looks wrong, the latter my impression is that there may have been intention for the snippets not to be inline like that?), but the rest is reasonably professional looking. Perhaps less severe, but the `levy_stable` bullet point is so verbose that it almost looks incorrectly formatted, but maybe that one is "ok".

<details>

image

image



</details>

File: doc/release/1.9.0-notes.rst [hunk 17]
```diff
@@ -70,46 +139,213 @@ New features
...
+- The ``extradoc`` keyword of ``rv_continuous``, ``rv_discrete`` and
+  ``rv_sample`` now raises a DeprecationWarning, after having been deprecated in
+  documentation only for a long time already.
+
+********************
+Expired Deprecations
+********************
+There is an ongoing effort to follow through on long-standing deprecations.
+The following previously deprecated features are affected:
+- Object arrays in sparse matrices now raise an error
```
Comment (WarrenWeckesser) [inline doc/release/1.9.0-notes.rst:250]: You need a blank line before the beginning of the bulleted list.  (I was going to link to the comment about this in the "missing bits" document, but apparently the link to that document in the rendered docs was broken by some recent reshuffling of the docs files. I see if I can fix that and submit a PR.)

Comment (WarrenWeckesser) [reply doc/release/1.9.0-notes.rst:250]: Here's the link to the relevant section of the docs from a PR that fixes the reference to the "missing bits" file: [Bulleted lists](https://output.circle-artifacts.com/output/job/7265df6e-f1f4-4e37-9ca6-2cd6a7b903f1/artifacts/0/html-scipyorg/dev/contributor/missing-bits.html#bulleted-lists)

**Comment by tupui** [comment]
@tylerjereddy before branching, there is an issue with the version switcher url (saying this here in case you wanted to branch before I make a PR tmr). Since the URL of the doc changed/got fixed, it also need to be adapted on the switcher. And we need to add 1.8.1.

File: doc/release/1.9.0-notes.rst [hunk 18]
```diff
@@ -70,46 +139,213 @@ New features
...
+  implementation and thus was accepting sparse arrays. Now it only works with
+  nDarrays. For sparse usage, `scipy.sparse.linalg.expm` needs to be used
+  explicitly.
+- The definition of `scipy.stats.circvar` has reverted to the one that is
+  standard in the literature; note that this is not the same as the square of
+  `scipy.stats.circstd`.
+- Remove inheritance to `QMCEngine` in `MultinomialQMC` and
+  `MultivariateNormalQMC`. It removes the methods `fast_forward` and `reset`.
+- Init of `MultinomialQMC` now require the number of trials with `n_trials`.
+  Hence, `MultinomialQMC.random` output has now the correct shape ``(n, pvals)``.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:312]:
```suggestion
  Hence, `MultinomialQMC.random` output has now the correct shape ``(n, pvals)``.
- Several function-specific warnings (``F_onewayConstantInputWarning``,
  ``F_onewayBadInputSizesWarning``, ``PearsonRConstantInputWarning``, 
  ``PearsonRNearConstantInputWarning``, ``SpearmanRConstantInputWarning``, and
  ``BootstrapDegenerateDistributionWarning``) have been replaced with more
  general warnings.
```

**Comment by tylerjereddy** [comment]
This is the last open milestoned PR, so branching is imminent. I'm iterating on some final relnotes updates/fixes locally now.

File: doc/release/1.9.0-notes.rst [hunk 19]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+  fixed. Note that ``stats.mode`` will now consume the input axis and return an
+  ndarray with the ``axis`` dimension removed.
+
+- Replaced implementation of `scipy.stats.ncf` with the implementation from
+  Boost for improved reliability.
+
+- Add a `bits` parameter to `scipy.stats.qmc.Sobol`. It allows to use from 0
+  to 64 bits to compute the sequence. Default is ``None`` which corresponds to
+  30 for backward compatibility. Using a higher value allow to sample more
+  points. Note: it does not correspond to the output type.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:192]:
```suggestion
  points. Note: ``bits`` does not affect the output dtype.
```

**Comment by tylerjereddy** [comment]
@mdhaber I committed your most recent suggestion, feel free to self-commit those small ones though, I hope to merge this and branch if I see that the CI is passing when I check a bit later.

File: doc/release/1.9.0-notes.rst [hunk 20]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
-
-
+- Due to collision with the shape parameter ``n`` of several distributions,
+  use of the distribution ``moment`` method with keyword argument ``n`` is
+  deprecated. Keyword ``n`` is replaced with keyword ``order``. 
+- Similarly, use of the distribution ``interval`` method with keyword arguments
+  ``alpha`` is deprecated. Keyword ``alpha`` is replaced with keyword
+  ``confidence``.
+- The ``'simplex'``, ``'revised simplex'``, and ``'interior-point'`` methods
+  of `scipy.optimize.linprog` are deprecated. Methods ``highs``, ``highs-ds``,
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:227]:
```suggestion
  of ``scipy.optimize.linprog`` are deprecated. Methods ``highs``, ``highs-ds``,
```
I'm not sure if this will link properly.

**Comment by tylerjereddy** [comment]
Updated the lists to account for last-minute gh-16241

File: doc/release/1.9.0-notes.rst [hunk 21]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- ``lobpcg`` performance improvements for small input cases
 
 `scipy.spatial` improvements
 ============================
-
-
-`scipy.special` improvements
-============================
+- Add a `order` parameter to `scipy.spatial.transform.Rotation.from_quat` and
+  `scipy.spatial.transform.Rotation.as_quat` to specify quaternion format.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:127]:
```suggestion
- Add an ``order` `parameter to `scipy.spatial.transform.Rotation.from_quat` 
  and `scipy.spatial.transform.Rotation.as_quat` to specify quaternion format.
```

File: doc/release/1.9.0-notes.rst [hunk 22]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
-
-
+- `scipy.stats.monte_carlo_test` performs one-sample Monte Carlo hypothesis
+  tests to assess whether a sample was drawn from a given distribution. Besides
+  reproducing the results of hypothesis tests like `scipy.stats.ks_1samp`,
+  `scipy.stats.normaltest`, and `scipy.stats.cramervonmises` without small sample
+  size limitations, it makes it possible to perform similar tests using arbitrary
+  statistics and distributions.
+
+- Several ``scipy.stats`` functions support new ``axis`` (integer or tuple of
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:139]:
```suggestion
- Several `scipy.stats` functions support new ``axis`` (integer or tuple of
```

File: doc/release/1.9.0-notes.rst [hunk 23]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+  integers) and ``nan_policy`` ('raise', 'omit', or 'propagate'), and
+  ``keepdims`` arguments.
+  These functions also support masked arrays as inputs, even if they do not have
+  a ``scipy.stats.mstats`` counterpart. Edge cases for multidimensional arrays,
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:143]:
```suggestion
  a `scipy.stats.mstats` counterpart. Edge cases for multidimensional arrays,
```

File: doc/release/1.9.0-notes.rst [hunk 24]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- Similarly, use of the distribution ``interval`` method with keyword arguments
+  ``alpha`` is deprecated. Keyword ``alpha`` is replaced with keyword
+  ``confidence``.
+- The ``'simplex'``, ``'revised simplex'``, and ``'interior-point'`` methods
+  of `scipy.optimize.linprog` are deprecated. Methods ``highs``, ``highs-ds``,
+  or ``highs-ipm`` should be used in new code.
+- Support for non-numeric arrays has been deprecated from ``stats.mode``.
+  ``pandas.DataFrame.mode`` can be used instead.
+- The function ``spatial.distance.kulsinski`` has been deprecated in favor
+  of ``spatial.distance.kulczynski1``.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:232]:
```suggestion
- The function `spatial.distance.kulsinski` has been deprecated in favor
  of `spatial.distance.kulczynski1`.
```

File: doc/release/1.9.0-notes.rst [hunk 25]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+********************
+Expired Deprecations
+********************
+There is an ongoing effort to follow through on long-standing deprecations.
+The following previously deprecated features are affected:
+
+- Object arrays in sparse matrices now raise an error
+- Inexact indices into sparse matrices now raise an error
+- Passing ``radius=None`` to ``spatial.SphericalVoronoi`` now raises an error
+  (not adding ``radius`` defaults to 1, as before)
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:251]:
```suggestion
- Passing ``radius=None`` to `scipy.spatial.SphericalVoronoi` now raises an
  error (not adding ``radius`` defaults to 1, as before)
```

File: doc/release/1.9.0-notes.rst [hunk 26]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- Several BSpline methods now raise an error if inputs have ``ndim > 1``.
+- The ``_rvs`` method of statistical distributions now requires a ``size``
+  parameter
+- Passing a ``fillvalue`` that cannot be cast to the output type in
+  ``signal.convolve2d`` now raises an error
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:256]:
```suggestion
  `scipy.signal.convolve2d` now raises an error
```

File: doc/release/1.9.0-notes.rst [hunk 27]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
 Backwards incompatible changes
 ******************************
 
+- Several `scipy.stats` functions now convert ``np.matrix`` to ``np.ndarray``s
+  before the calculation is performed. In this case, the output will be a scalar
+  or ``np.ndarray`` of appropriate shape rather than a 2D ``np.matrix``.
+  Similarly, while masked elements of masked arrays are still ignored, the
+  output will be a scalar or ``np.ndarray`` rather than a masked array with
+  ``mask=False``.
+- The default method of ``scipy.optimize.linprog`` is now ``'highs'``, not
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:292]:
```suggestion
- The default method of `scipy.optimize.linprog` is now ``'highs'``, not
```

File: doc/release/1.9.0-notes.rst [hunk 28]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- Object arrays in sparse matrices now raise an error
+- Inexact indices into sparse matrices now raise an error
+- Passing ``radius=None`` to ``spatial.SphericalVoronoi`` now raises an error
+  (not adding ``radius`` defaults to 1, as before)
+- Several BSpline methods now raise an error if inputs have ``ndim > 1``.
+- The ``_rvs`` method of statistical distributions now requires a ``size``
+  parameter
+- Passing a ``fillvalue`` that cannot be cast to the output type in
+  ``signal.convolve2d`` now raises an error
+- The ``spatial.distance`` now enforce that the input vectors are
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:257]:
```suggestion
- `scipy.spatial.distance` now enforces that the input vectors are
```

File: doc/release/1.9.0-notes.rst [hunk 29]
```diff
@@ -26,90 +26,329 @@ For running on PyPy, PyPy3 6.0+ is required.
...
+- Add a ``dtype`` parameter to `scipy.sparce.csgraph.laplacian` for type casting.
+- Add a ``symmetrized`` parameter to `scipy.sparce.csgraph.laplacian` to produce
+  symmetric Laplacian for directed graphs.
+- Add a ``form`` parameter to `scipy.sparce.csgraph.laplacian` taking one of the
+  three values: ``array``, or ``function``, or ``lo`` determining the format of
+  the output Laplacian:
+  * ``array`` is a numpy array (backward compatible default);
+  * ``function`` is a pointer to a lambda-function evaluating the
+  Laplacian-vector or Laplacian-matrix product;
+  * ``lo`` results in the format of the `LinearOperator`.
```
Comment (mdhaber) [inline doc/release/1.9.0-notes.rst:118]:
```suggestion
  * ``lo`` results in the format of the ``LinearOperator``.
```

**Review by mdhaber — APPROVED** [review]
I'm sure there are many single VS double backtick issues still present, but I should probably stop so as not to hold up release.

**Comment by tylerjereddy** [comment]
Ok, the docs are building/passing in CI so in it goes. I don't think we need to flush Azure an Nth time for release notes typo fixes, etc.

File: doc/release/1.9.0-notes.rst [hunk 30]
```diff
@@ -152,14 +391,674 @@ For more information on the introduction of Meson support in SciPy, see
...
+* David Zwicker (1) +
+
+A total of 152 people contributed to this release.
+People with a "+" by their names contributed a patch for the first time.
+This list of names is automatically generated, and may not be fully complete.
 
 
 ***********************
 Issues closed for 1.9.0
 ***********************
```
[the remark this answers is not in the document]
Comment (rgommers) [reply doc/release/1.9.0-notes.rst:554]: Indeed, see `tools/gh_lists.py`.

Grabbing issues based on date of closure isn't going to work, because a lot are closed because fixed in older releases, invalid reports, etc.

As of now, you should set milestones on issues that you close. A way to improve the script could be to parse commit messages for "Closes xxx", but even that isn't going to be foolproof since I often see that people fail to add that info to commit messages and only add it to PR descriptions.

[the remark this answers is not in the document]
Comment (ev-br) [reply doc/release/1.9.0-notes.rst:554]: A possible automation could be to auto-add a milestone to an issue which is closed by a PR  with a milestone. Maybe that's a feature request to github or maybe it's possible to do with their api.

[the remark this answers is not in the document]
Comment (mdhaber) [reply doc/release/1.9.0-notes.rst:554]: I started to do this for issues I was involved with. I thought I should make sure I'm doing the right thing before going further. For instance, I added the milestone to gh-14622 because it was a legitimate issue that was closed by a PR, whereas I did not add a milestone to gh-15160 because it was reported due to a misunderstanding. LMK if I should keep going this way.

[the remark this answers is not in the document]
Comment (rgommers) [reply doc/release/1.9.0-notes.rst:554]: Yes, that sounds right, thanks Matt!

[the remark this answers is not in the document]
Comment (mdhaber) [reply doc/release/1.9.0-notes.rst:554]: OK, done. I added a lot of milestones. 
I accidentally created a 1.90 milestone in the process. Do you know how to remove that from the list?

[the remark this answers is not in the document]
Comment (rgommers) [reply doc/release/1.9.0-notes.rst:554]: Thanks Matt. I deleted the 1.90 milestone.
